core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -595,7 +593,7 @@ private[spark] object Utils extends Logging {
     if (lowerSrc.endsWith(".jar")) {
       RunJar.unJar(source, dest, RunJar.MATCH_ANY)
     } else if (lowerSrc.endsWith(".zip")) {
-      FileUtil.unZip(source, dest)
+      unZip(source.getAbsoluteFile, dest)
```

Comment:
Why do we need a different code path for unzipping only here and not elsewhere?
I'm also concerned about adding the platform dependency on unzip just for this.
Why does the exec bit need to be set - wouldn't you run a script as `python script.py` anyway to be more sure about the python being used?

Reply:
This is the issue of my premis, https://issues.apache.org/jira/browse/SPARK-37677, The submission method I mentioned is not supported after the zip file is unzipped.

Reply:
Do you mean that the absolute path of the source file is different from other decompression methods? This can use relative paths. I use absolute paths for easy understanding at that time.

Reply:
No, there's an existing unZip method. At least, why not change that?

Reply:
The existing zip decompression method can not retain the original permission information of the file, but can only retain the read-write permission, so the user's zip file has no execution permission after decompression. The current example is that there is no python or python3 command after decompression.

Reply:
My point is, would that be helpful in all unzip scenarios? rather than have two different implementations
Adding a dependency on the OS package is probably OK but not sure
Can you add python/python3 to the execution? that seems more standard but not sure if it is possible

Reply:
https://github.com/apache/spark/pull/35278, This PR is to add execution permissions to python/python3, but the suggestion is to retain the permissions of the original file instead of adding permissions.